arangod/Aql/ExecutionBlockImpl.cpp

```diff
@@ -63,25 +63,40 @@
 using namespace arangodb;
 using namespace arangodb::aql;
 
+
 template <class Executor>
 ExecutionBlockImpl<Executor>::ExecutionBlockImpl(ExecutionEngine* engine,
                                                  ExecutionNode const* node,
                                                  typename Executor::Infos&& infos)
     : ExecutionBlock(engine, node),
-      _blockFetcher(_dependencies, _engine->itemBlockManager(),
+      _blockFetcher(_dependencies, engine->itemBlockManager(),
                     infos.getInputRegisters(), infos.numberOfInputRegisters()),
       _rowFetcher(_blockFetcher),
       _infos(std::move(infos)),
       _executor(_rowFetcher, _infos),
       _outputItemRow(),
-      _query(*engine->getQuery()) {}
+      _query(*engine->getQuery()),
+      _engine(engine),
+      _trx(engine->getQuery()->trx()),
+      _pos(0) {
+  TRI_ASSERT(_trx != nullptr);
+
+  // already insert ourselves into the statistics results
+  if (_profile >= PROFILE_LEVEL_BLOCKS) {
+    _engine->_stats.nodes.emplace(node->id(), ExecutionStats::Node());
+  }
+}
 
 template <class Executor>
-ExecutionBlockImpl<Executor>::~ExecutionBlockImpl() {}
+ExecutionBlockImpl<Executor>::~ExecutionBlockImpl() {
+  for (auto& it : _buffer) {
+    delete it;
+  }
+}
 
 template <class Executor>
 std::pair<ExecutionState, std::unique_ptr<AqlItemBlock>> ExecutionBlockImpl<Executor>::getSome(size_t atMost) {
-  traceGetSomeBegin(atMost);
+  ExecutionBlock::traceGetSomeBegin(atMost);
   auto result = getSomeWithoutTrace(atMost);
   return traceGetSomeEnd(result.first, std::move(result.second));
```

Comment:
This should call `ExecutionBlock::traceGetSomeEnd`, and `traceGetSomeEnd` here should be removed. Also, when both `traceGetSomeBegin` and `traceGetSomeEnd` are removed in this class, you can just call `traceGetSomeBegin` and `traceGetSomeEnd` without specifying `ExecutionBlock::`.